### PR TITLE
update the __repr__ method of the solution

### PR DESCRIPTION
**Pull request by geoffreyp**
new representation of solutions : 
Solution(F(x)=[..., ...., ....] ; x=[..,..,..,..,..,..])

### CODE REVIEW

File: moead_framework/solution/one_dimension_solution.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def __init__(self, solution, f=None):
         self.distance = 0
 
     def __repr__(self):
-        return str(self.F)
+        return "Solution(F(x)=" + str(self.F) + " ; x=" +str(list(self.solution)) + ")"
```
Comment (sjvrijn) [inline moead_framework/solution/one_dimension_solution.py:24]: This does not conform to the specification for `__repr__` as stated in the [Python documentation](https://docs.python.org/3/reference/datamodel.html#object.__repr__).
For `__repr__` we would expect `f'OneDimensionSolution(solution={self.solution}, f={self.F})'`, which can be parsed by `eval()` and is intended for use by developers.

The `__str__` method is intended to provide a more user-friendly string representation, which can be the format you specify in this commit.

**Review by sjvrijn — CHANGES_REQUESTED** [review]

File: moead_framework/solution/one_dimension_solution.py (hunk 1 above)
Comment (geoffreyp) [reply moead_framework/solution/one_dimension_solution.py:24]: I fixed it in the commit d099d4a in this PR

Comment (sjvrijn) [reply moead_framework/solution/one_dimension_solution.py:24]: My suggestion was using the f-string syntax of Python 3.6+: https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings

File: moead_framework/solution/one_dimension_solution.py [hunk 2]
```diff
@@ -20,9 +20,12 @@ def __init__(self, solution, f=None):
         super().__init__(solution, f)
         self.distance = 0
 
-    def __repr__(self):
+    def __str__(self):
         return "Solution(F(x)=" + str(self.F) + " ; x=" +str(list(self.solution)) + ")"
 
+    def __repr__(self):
+        return "f'OneDimensionSolution(solution={" + str(list(self.solution)) + "}, f={" + str(self.F) + "})'"
```
Comment (sjvrijn) [inline moead_framework/solution/one_dimension_solution.py:27]: `return f'OneDimensionSolution(solution={list(self.solution)}, f={self.F})'`
This takes care of the formatting in Python 3.6+, without needing `str()` calls and manually joining substrings with `+`

**Review by sjvrijn — CHANGES_REQUESTED** [review]

File: moead_framework/solution/one_dimension_solution.py (hunk 2 above)
Comment (geoffreyp) [reply moead_framework/solution/one_dimension_solution.py]: I fixed it 3485349

File: moead_framework/solution/one_dimension_solution.py [hunk 3]
```diff
@@ -20,8 +20,11 @@ def __init__(self, solution, f=None):
         super().__init__(solution, f)
         self.distance = 0
 
+    def __str__(self):
+        return "Solution(F(x)=" + str(self.F) + " ; x=" +str(list(self.solution)) + ")"
+
     def __repr__(self):
-        return str(self.F)
+        return f'OneDimensionSolution(solution={self.solution}, f={self.F})'
```
Comment (sjvrijn) [inline moead_framework/solution/one_dimension_solution.py:27]: One final request: wrap `self.solution` in a `list()` call, so `f'OneDimensionSolution(solution={list(self.solution)}, f={self.F})'`

An `eval(repr(x))` round-trip works fine without the `list()` call if `self.solution` is a list, but if it's a numpy array, this errors:

```python
>>> tmp = OneDimensionSolution([1,2,3], f=[2,3])
>>> eval(repr(tmp)).__dict__ == tmp.__dict__  # comparing on __dict__ for a quick equality check 
True
>>> tmp = OneDimensionSolution(np.array([1,2,3]), f=[2,3])
>>> eval(repr(tmp)).__dict__ == tmp.__dict__
Traceback (most recent call last):
    [...]
    OneDimensionSolution(solution=[1 2 3], f=[2, 3])
                         ^
SyntaxError: expression cannot contain assignment, perhaps you meant "=="?
```

The reason is that numpy arrays are by default displayed without commas between their elements. This is avoided with a `list()` call:
```python
>>> x = np.array([1, 2, 3])
>>> print(f'array x is {x}')
array x is [1 2 3]
             ^ ^
        # just spaces, no commas
>>> print(f'array x is {list(x)}')
array x is [1, 2, 3]
             ^  ^
       # now with commas
```

**Review by sjvrijn — CHANGES_REQUESTED** [review]
Great, just one final request, then it's perfect

File: moead_framework/solution/one_dimension_solution.py (hunk 3 above)
Comment (geoffreyp) [reply moead_framework/solution/one_dimension_solution.py]: it's done with 919c4b1

**Review by sjvrijn — APPROVED** [review]

**Pull request merged**